src/librustc/mir/interpret/value.rs

```diff
@@ -17,8 +17,8 @@ pub struct RawConst<'tcx> {
     pub ty: Ty<'tcx>,
 }
 
-/// Represents a constant value in Rust. `Scalar` and `ScalarPair` are optimizations that
-/// match the `LocalState` optimizations for easy conversions between `Value` and `ConstValue`.
+/// Represents a constant value in Rust. `Scalar` and `Slice` are optimizations that
+/// match the `LocalState` optimizations for easy conversions between `Operand` and `ConstValue`.
```

Comment:
Oh this comment is entirely outdated. The conversion is actually not what we're optimizing for anymore. These two variants are optimizations for array length computations, enum discriminants and the pattern matching logic